R/checkPythonDeps.R

```diff
@@ -0,0 +1,350 @@
+#' checkPythonDeps.R
+#'
+#' Verify the availability of REMIND's Python dependencies on the host system. Exposes two functions:
+#' \code{checkPythonRequirements} and \code{checkPythonDeps}. While \code{checkPythonRequirements} reads a pip-style
+#' requirements file and compares the required packages with the installed ones, \code{checkPythonDeps} checks if the
+#' required Python dependencies can actually be imported in the Python environment provided.
+#'
+#' @author Tonn Rüter
+
+# ---------------------
+# Utility Functions
+# ---------------------
+
+#' Create Python Version
+#'
+#' This function creates a named list representing a Python version string from its components.
+#'
+#' @param major Major version number
+#' @param minor Minor version number
+#' @param patch Patch version number
+#' @param releaseType Release type (e.g. alpha, beta, rc)
+#' @param releaseVersion Release version number
+#' @return A named list representing a Python version string
+#' @importFrom reticulate import py_list_packages
+#' @importFrom stringr regex str_match
+#' @importFrom purrr map map_chr pmap
+#' @importFrom utils strcapture
+createPythonVersion <- function(major = "", minor = "", patch = "", releaseType = "", releaseVersion = "") {
+  # Store the components in a named list
+  return(list(
+    major = as.integer(major),
+    minor = as.integer(minor),
+    patch = as.integer(patch),
+    releaseType = as.character(releaseType),
+    releaseVersion = as.integer(releaseVersion)
+  ))
+}
+
+#' Print Python version string
+#'
+#' This function creates a string representation of a Python version.
+#'
+#' @param pythonVerion A named list representing the Python version.
+#' @return A string representation of the Python version.
+printPythonVersion <- function(pythonVerion) {
+  return(paste0(
+    ifelse(is.na(pythonVerion$major), "", as.character(pythonVerion$major)),
```

Comment:
```suggestion
    if (is.na(pythonVerion$major)) "" else as.character(pythonVerion$major),
```
While this works ifelse is actually discouraged for vectors of length 1 according to https://rdrr.io/r/base/ifelse.html :shrug: